commit 8268821b434d1308d083454fb681d80176cf352b
Author: Xiao Ni <xni@redhat.com>
Date:   Fri Mar 17 19:55:42 2017 +0800

    mdadm: Add Wimplicit-fallthrough=0 in Makefile
    
    There are many errors like 'error: this statement may fall through'.
    But the logic is right. So add the flag Wimplicit-fallthrough=0
    to disable the error messages. The method I use is from
    https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
    #index-Wimplicit-fallthrough-375
    
    Signed-off-by: Xiao Ni <xni@redhat.com>
    Signed-off-by: Jes Sorensen <Jes.Sorensen@gmail.com>

diff --git a/Makefile b/Makefile
index a6f464c..d1a6ac4 100644
--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,11 @@ ifdef WARN_UNUSED
 CWFLAGS += -Wp,-D_FORTIFY_SOURCE=2 -O3
 endif
 
+FALLTHROUGH := $(shell gcc -v --help 2>&1 | grep "implicit-fallthrough" | wc -l)
+ifneq "$(FALLTHROUGH)"  "0"
+CWFLAGS += -Wimplicit-fallthrough=0
+endif
+
 ifdef DEBIAN
 CPPFLAGS += -DDEBIAN
 endif
